yahp/utils/json_schema_helpers.py

```diff
@@ -0,0 +1,88 @@
+import inspect
+from enum import Enum
+from typing import Any, Dict
+
+from yahp.utils import type_helpers
+
+
+def get_registry_json_schema(f_type: type_helpers.HparamsType, registry: Dict[str, Any]):
+    """Convert type into corresponding JSON Schema. As the given name is in the `hparams_registry`,
+    create objects for each possible entry in the registry and treat as union type.
+    """
+    res = {'anyOf': []}
+    for key, value in registry.items():
+        res['anyOf'].append({
+            'type': 'object',
+            'properties': {
+                key: get_type_json_schema(type_helpers.HparamsType(value))
+            },
+            'additionalProperties': False,
+        })
+    return check_for_list_and_optional(f_type, res)
+
+
+def get_type_json_schema(f_type: type_helpers.HparamsType):
+    """Convert type into corresponding JSON Schema.
+    """
+    # Import inside function to resolve circular dependencies
+    from yahp.auto_hparams import ensure_hparams_cls
+
+    res = {}
+
+    # Union
+    if len(f_type.types) > 1:
+        # Add all union types using anyOf
+        res = {'anyOf': []}
+        for union_type in f_type.types:
+            res['anyOf'].append(get_type_json_schema(type_helpers.HparamsType(union_type)))
+    # Primitive Types
+    elif f_type.types[0] is str:
+        res = {'type': 'string'}
+    elif f_type.types[0] is bool:
+        res = {'type': 'boolean'}
+    elif f_type.types[0] is int:
+        res = {'type': 'integer'}
+    elif f_type.types[0] is float:
+        res = {'type': 'number'}
+    # Enum
+    elif inspect.isclass(f_type.types[0]) and issubclass(f_type.types[0], Enum):
```

Comment:
```suggestion
    elif ftype.is_enum
```
you might need to add an `assert isinstane(ftype.type, Enum)` below.
(Note: I don't remember how HparamsType handles `Union[str, Enum]` -- might need to condition on `ftype.types` instead of `ftype.type`)

Reply:
`Union[str, Enum]` raises a type error. Is enum checks `assert issubclass(self.type, Enum)` so I think this should be fine. Note that all union types are already recursively unpacked by the first `if` clause, so at this point we know it's not a union